client/state/data-layer/wpcom/login-2fa/index.js

```diff
@@ -20,71 +14,91 @@ import {
+ * Dispatch action to perform http request
  *
- * @param {Object}   store  Global redux store
- * @returns {Promise}		Promise of result from the API
+ * @param {Object}	store  Global redux store
+ * @param {Object}	action dispathced action
  */
-const doAppPushRequest = ( store ) => {
-	return request.post( 'https://wordpress.com/wp-login.php?action=two-step-authentication-endpoint' )
-		.withCredentials()
-		.set( 'Content-Type', 'application/x-www-form-urlencoded' )
-		.accept( 'application/json' )
-		.send( {
+const requestTwoFactorPushNotificationStatus = ( store, action ) => {
+	store.dispatch( rawHttp( {
+		url: 'https://wordpress.com/wp-login.php?action=two-step-authentication-endpoint',
+		method: 'POST',
+		headers: [
+			[ 'Content-Type', 'application/x-www-form-urlencoded' ],
+		],
+		withCredentials: true,
+		body: {
 			user_id: getTwoFactorUserId( store.getState() ),
 			two_step_nonce: getTwoFactorAuthNonce( store.getState() ),
 			remember_me: getTwoFactorRememberMe( store.getState() ),
 			two_step_push_token: getTwoFactorPushToken( store.getState() ),
 			client_id: config( 'wpcom_signup_id' ),
 			client_secret: config( 'wpcom_signup_key' ),
-		} ).then( () => {
-			store.dispatch( { type: TWO_FACTOR_AUTHENTICATION_PUSH_POLL_COMPLETED } );
-		} ).catch( error => {
-			store.dispatch( { type: TWO_FACTOR_AUTHENTICATION_PUSH_UPDATE_NONCE, twoStepNonce: error.response.body.data.two_step_nonce } );
-			return Promise.reject( error );
-		} );
+		}
+	}, action ) );
 };
 
 /***
- * Polling the login API for the status of the push notification.
- * The polling will stop on success or when store's polling progress
- * state changes to `false`
+ * Receive data from the two factor push notification status http request
  *
- * @param {Object}   store  Global redux store
+ * @param {Function}	dispatch redux store dispatch function
+ * @returns {*} dispatch result
  */
-const doAppPushPolling = store => {
-	let retryCount = 0;
-	const retry = () => {
-		// if polling was stopped or not in progress - stop
-		if ( ! getTwoFactorPushPollInProgress( store.getState() ) ) {
-			return;
-		}
+const receivedTwoFactorPushNotificationApproved = ( { dispatch } ) =>
+	dispatch( { type: TWO_FACTOR_AUTHENTICATION_PUSH_POLL_COMPLETED } );
 
-		retryCount++;
-		setTimeout(
-			() => doAppPushRequest( store ).catch( retry ),
-			( POLL_APP_PUSH_INTERVAL_SECONDS + Math.floor( retryCount / 10 ) ) * 1000 // backoff lineary
-		);
-	};
+/***
+ * Receive error from the two factor push notification status http request
+ *
+ * @param {Object}	store  Global redux store
+ * @param {Object}	action dispathced action
+ * @param {Function}	next continue dispatch function
+ * @param {Object}	error the error object
+ */
+const receivedTwoFactorPushNotificationError = ( store, action, next, error ) => {
+	store.dispatch( { type: TWO_FACTOR_AUTHENTICATION_PUSH_UPDATE_NONCE, twoStepNonce: error.response.body.data.two_step_nonce } );
```

Comment:
let's be careful here since we're introducing a possible `TypeError` into production. a safe-defaulting getter might be more appropriate than direct nested property access because if the request fails for any unexpected reason it's not likely to have the required shape in the `error` response and this will _throw_.

```js
import { get } from 'lodash';

twoStepNonce: get( error, 'response.body.data.two_step_nonce', '' ) // defaults to `null` if no third parameter given
```